client/lib/purchases/test/store-test.js

```diff
@@ -25,11 +51,72 @@ describe( 'Purchases Store', () => {
 				purchases: [ { id: 2 }, { id: 3 } ]
 			} );
 
-			assert( find( PurchasesStore.get().data, { id: 1 } ) );
-			assert( find( PurchasesStore.get().data, { id: 2 } ) );
-			assert( find( PurchasesStore.get().data, { id: 3 } ) );
+			expect( PurchasesStore.get() ).to.be.eql( {
+				data: [ { id: 2 }, { id: 3 }, { id: 1 } ],
+				error: null,
+				hasLoadedFromServer: true,
+				isFetching: false
+			} );
 
 			done();
 		} );
 	} );
+
+	it( 'should return an object with original purchase when cancelation of private registration is triggered', () => {
+		Dispatcher.handleServerAction( {
+			type: actionTypes.PURCHASES_PRIVATE_REGISTRATION_CANCEL,
+			purchaseId: 2
+		} );
+
+		expect( PurchasesStore.getByPurchaseId( 2 ) ).to.be.eql( {
+			data: {
+				id: 2
+			},
+			error: null,
+			hasLoadedFromServer: true,
+			isFetching: false
+		} );
+	} );
+
+	it( 'should return an object with original purchase and error message when cancelation of private registration failed', () => {
+		Dispatcher.handleServerAction( {
+			type: actionTypes.PURCHASES_PRIVATE_REGISTRATION_CANCEL_FAILED,
+			error: 'Unable to fetch stored cards',
+			purchaseId: 2
+		} );
+
+		expect( PurchasesStore.getByPurchaseId( 2 ) ).to.be.eql( {
+			data: {
+				id: 2,
+				error: 'Unable to fetch stored cards'
+			},
+			error: null,
+			hasLoadedFromServer: true,
+			isFetching: false
+		} );
+	} );
+
+	it( 'should return an object with updated purchase when cancelation of private registration completed', () => {
+		Dispatcher.handleServerAction( {
+			type: actionTypes.PURCHASES_PRIVATE_REGISTRATION_CANCEL_COMPLETED,
+			purchase: {
+				amount: 2200,
+				error: null,
```

Comment:
Well I think we should indeed revisit how we structure data in our stores, and how we access it. Unit tests are a good way to highlight this type of issues - take a look at the object returned below by `PurchasesStore.getByPurchaseId( 2 )`: there are two `error` fields. So yes I agree, one of them must die :).